reduction/lr_reduction/event_reduction.py

```diff
@@ -1251,3 +1258,30 @@ def compute_resolution(ws, default_dq=0.027, theta=None, q_summing=False):
     slit_distance = s1_sample_distance - sample_si_distance
     dq_over_q = s1h / slit_distance / theta
     return dq_over_q
+
+def compute_wavelength_resolution(ws):
+    """
+    Compute the wavelength resolution from the meta data.
+
+    Parameters
+    ----------
+    ws : mantid.api.Workspace
+        Mantid workspace to extract correction meta-data from.
+
+    Returns
+    -------
+    tuple of List[Float]
+        (wavelength, d_lambda):
+            wavelength: the fitted wavelength values
+            d_lambda: the difference between wavelength and the fit
+    """
+    settings = read_settings(ws)
+
+    out = api.EvaluateFunction(Function=settings.wavelength_resolution_function,
+                         InputWorkspace=ws,
```

Comment:
ask Becky if `ws` is going to contain only one spectrum and if so, put a check here to test that's the case